MAUI Documentation Cleanup

michaelstonis opened the pull request:
## Summary

* Renames configuration management to App Settings. This better reflects the use case for this section.
* Capitalization updates
* Updates section on dependency service registration and building to be more explicit.

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/architecture/maui/accessing-remote-data.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/accessing-remote-data.md) | [Accessing Remote Data](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/accessing-remote-data?branch=pr-en-us-41184) |
| [docs/architecture/maui/app-settings-management.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/app-settings-management.md) | [Application Settings Management](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/app-settings-management?branch=pr-en-us-41184) |
| [docs/architecture/maui/authentication-and-authorization.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/authentication-and-authorization.md) | [Authentication and Authorization](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/authentication-and-authorization?branch=pr-en-us-41184) |
| [docs/architecture/maui/communicating-between-components.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/communicating-between-components.md) | [docs/architecture/maui/communicating-between-components](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/communicating-between-components?branch=pr-en-us-41184) |
| [docs/architecture/maui/dependency-injection.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/dependency-injection.md) | [Dependency injection](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/dependency-injection?branch=pr-en-us-41184) |
| [docs/architecture/maui/index.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/index.md) | [Enterprise Application Patterns Using .NET MAUI](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/index?branch=pr-en-us-41184) |
| [docs/architecture/maui/introduction.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/introduction.md) | [Introduction to .NET MAUI](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/introduction?branch=pr-en-us-41184) |
| [docs/architecture/maui/micro-services.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/micro-services.md) | [Containerized Microservices](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/micro-services?branch=pr-en-us-41184) |
| [docs/architecture/maui/mvvm-community-toolkit-features.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/mvvm-community-toolkit-features.md) | [docs/architecture/maui/mvvm-community-toolkit-features](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/mvvm-community-toolkit-features?branch=pr-en-us-41184) |
| [docs/architecture/maui/mvvm.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/mvvm.md) | [Model-View-ViewModel (MVVM)](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/mvvm?branch=pr-en-us-41184) |
| [docs/architecture/maui/navigation.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/navigation.md) | [Navigation](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/navigation?branch=pr-en-us-41184) |
| [docs/architecture/maui/preface.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/preface.md) | [Purpose](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/preface?branch=pr-en-us-41184) |
| [docs/architecture/maui/unit-testing.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/unit-testing.md) | [docs/architecture/maui/unit-testing](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/unit-testing?branch=pr-en-us-41184) |
| [docs/architecture/maui/validation.md](https://github.com/dotnet/docs/blob/59889dfb959b8da37d978e3024d13175797565d6/docs/architecture/maui/validation.md) | [Validation](https://review.learn.microsoft.com/en-us/dotnet/architecture/maui/validation?branch=pr-en-us-41184) |

</details>


<!-- PREVIEW-TABLE-END -->

michaelstonis committed:
rename configuration management to app settings

Rename configuration management to app settings

Renamed all instances of 'configuration management' to 'app settings' in the documentation. This includes changes in file names, titles, and references within the text. The table of contents was also updated to reflect this change.

michaelstonis committed:
fix capitalization

michaelstonis committed:
update to be less ambiguous

IEvangelist commented:
We need a redirection for the file name change. Since the file name also serves as the route, we need to configure where the old route leads to now. You'll update this file:

- https://github.com/dotnet/docs/blob/main/.openpublishing.redirection.architecture.json

You'll add an entry to the bottom like this:

```json
        {
            "source_path_from_root": "/docs/architecture/maui/configuration-management.md",
            "redirect_url": "/dotnet/architecture/maui/app-settings-management"
        }
```

michaelstonis committed:
add redirect for app settings

michaelstonis commented:
Added redirect for the app settings chapter rename.

jamesmontemagno commented on .openpublishing.redirection.architecture.json:
```diff
@@ -812,6 +812,10 @@
+        },
+        {
+            "source_path_from_root": "/docs/architecture/maui/configuration-management.md",
+            "redirect_url": "/dotnet/architecture/maui/app-settings-management.md"
```
```suggestion
            "redirect_url": "/dotnet/architecture/maui/app-settings-management"
```

jamesmontemagno committed:
Apply suggestions from code review

jamesmontemagno committed:
Update toc.yml

jamesmontemagno committed:
Update toc.yml

jamesmontemagno committed:
updated dates

jamesmontemagno commented:
@michaelstonis can we update images. I updated versions and renamed to eShop

jamesmontemagno commented:
@IEvangelist let's wait to merge until we get the updates from Michael above

michaelstonis committed:
Updating from eShopOnContainers to eShop

Updated references from "eShopOnContainers" to "eShop". Changes include updates in documentation and namespace adjustments in code. This modification ensures consistency across the application's naming convention.

michaelstonis committed:
Merge branch 'maui-net8/misc-docs-cleanup' of https://github.com/michaelstonis/docs into maui-net8/misc-docs-cleanup

michaelstonis committed:
update app screenshots

jamesmontemagno commented on docs/architecture/maui/preface.md:
```diff
@@ -14,7 +14,7 @@ This eBook provides guidance on building cross-platform enterprise apps using .N
 The guide provides architectural guidance for developing adaptable, maintainable, and testable .NET MAUI enterprise apps. Guidance is provided on how to implement MVVM, dependency injection, navigation, validation, and configuration management, while maintaining loose coupling. In addition, there's also guidance on performing authentication and authorization with IdentityServer, accessing data fr ...
 
-The guide comes with source code for the [eShopOnContainers multi-platform app](https://github.com/dotnet-architecture/eShopOnContainers/tree/main/src/Mobile), and source code for the [eShopOnContainers reference app](https://github.com/dotnet-architecture/eShopOnContainers). The eShopOnContainers multi-platform app is a cross-platform enterprise app developed using .NET MAUI, which connects to a ...
+The guide comes with source code for the [eShop multi-platform app](https://github.com/dotnet-architecture/eShop/tree/main/src/Mobile), and source code for the [eShop reference app](https://github.com/dotnet-architecture/eShop). The eShop multi-platform app is a cross-platform enterprise app developed using .NET MAUI, which connects to a series of containerized microservices known as the eShop re ...
```
```suggestion
The guide comes with source code for the [eShop multi-platform app](https://github.com/dotnet/eShop/tree/main/src/ClientApp), and source code for the [eShop reference app](https://github.com/dotnet/eShop). The eShop multi-platform app is a cross-platform enterprise app developed using .NET MAUI, which connects to a series of containerized microservices known as the eShop reference app. However, the eShop multi-platform app can be configured to consume data from mock services for those who wish to avoid deploying the containerized microservices.
```

jamesmontemagno committed:
Update docs/architecture/maui/preface.md

michaelstonis committed:
updated images with backgrounds and latest app design

michaelstonis committed:
add image background

michaelstonis committed:
Merge remote-tracking branch 'upstream/main' into maui-net8/misc-docs-cleanup

jamesmontemagno commented:
@IEvangelist any way to resolve conflict, else good

jamesmontemagno commented:
Replacing with https://github.com/dotnet/docs/pull/41325 since something is off with the system here....

michaelstonis committed:
Merge remote-tracking branch 'upstream/main' into maui-net8/misc-docs-cleanup

jamesmontemagno commented:
had some additional edits in mine: https://github.com/dotnet/docs/pull/41325 so will pull that one in instead if that is cool.
